datapipelines/src/main/kotlin/io/customer/sdk/CustomerIOBuilder.kt

```diff
@@ -59,6 +59,9 @@ class CustomerIOBuilder(
     // Configuration options required for migration from earlier versions
     private var migrationSiteId: String? = null
 
+    // Overrides the default DataPipelines module instance with custom implementation
+    private var overrideDataPipelineInstance: ((DataPipelinesModuleConfig) -> CustomerIO)? = null
```

Comment:
I think if we were to utilize delegate from DiGraph we won't have to override manually?

Reply:
Yes, this is a good suggestion. The only problem is, we need `DataPipelinesModuleConfig` to create test instance. We can think of options to solve this problem though, but I think using `DiGraph` can simplify a lot of code in this PR. I'll try this out and get back soon.

Reply:
I added some updates to `SDKComponent` as we discussed on call earlier, let me know if there are any concerns.